controller/admin_controller.php

```diff
@@ -0,0 +1,288 @@
+<?php
+/**
+ *
+ * Ideas extension for the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\ideas\controller;
+
+/**
+* Admin controller
+*/
+class admin_controller implements admin_interface
+{
+	/** @var \phpbb\config\config */
+	protected $config;
+
+	/** @var \phpbb\db\driver\driver_interface */
+	protected $db;
+
+	/** @var \phpbb\language\language */
+	protected $language;
+
+	/** @var \phpbb\log\log */
+	protected $log;
+
+	/** @var \phpbb\request\request */
+	protected $request;
+
+	/** @var \phpbb\template\template */
+	protected $template;
+
+	/** @var \phpbb\user */
+	protected $user;
+
+	/** @var array */
+	protected $cfg_array;
+
+	/** @var array */
+	protected $new_config;
+
+	/** @var string */
+	protected $phpbb_root_path;
+
+	/** @var string */
+	protected $php_ext;
+
+	/** @var string */
+	public $u_action;
+
+	/**
+	* Constructor
+	*
+	* @param \phpbb\config\config              $config               Config object
+	* @param \phpbb\db\driver\driver_interface $db                   Database object
+	* @param \phpbb\language\language          $language             Language object
+	* @param \phpbb\log\log                    $log                  Log object
+	* @param \phpbb\request\request            $request              Request object
+	* @param \phpbb\template\template          $template             Template object
+	* @param \phpbb\user                       $user                 User object
+	* @param string                            $root_path            phpBB root path
+	* @param string                            $php_ext              php_ext
+	* @access public
+	*/
+	public function __construct(\phpbb\config\config $config, \phpbb\db\driver\driver_interface $db, \phpbb\language\language $language, \phpbb\log\log $phpbb_log, \phpbb\request\request $request, \phpbb\template\template $template, \phpbb\user $user, $root_path, $php_ext)
+	{
+		$this->config = $config;
+		$this->db = $db;
+		$this->language = $language;
+		$this->log = $phpbb_log;
+		$this->request = $request;
+		$this->template = $template;
+		$this->user = $user;
+		$this->phpbb_root_path = $root_path;
+		$this->php_ext = $php_ext;
+	}
+
+	/**
+	* Display the options a user can configure for this extension
+	*
+	* @return null
+	* @access public
+	*/
+	public function display_options($id, $mode)
```

Comment:
Are these needed? They're not used, nor passed in when the function is called from the module.